client/blocks/comment-detail/comment-detail-author.jsx

```diff
@@ -59,9 +70,56 @@ export class CommentDetailAuthor extends Component {
 		gmtOffset( this.props.site )
 	).format( 'll LT' );
 
+	showMoreInfo = () => !! this.props.authorEmail || !! this.props.authorIp || !! this.props.authorUrl;
+
+	toggleBlockUser = () => {
+		const {
+			authorEmail,
+			authorId,
+			authorIsBlocked,
+			siteBlacklist,
+			commentId,
+			translate,
+			updateBlacklist,
+		} = this.props;
+
+		const noticeOptions = {
+			duration: 5000,
+			id: `comment-notice-${ commentId }`,
+			isPersistent: true,
+		};
+
+		const analytics = {
+			action: authorIsBlocked ? 'unblock_user' : 'block_user',
+			user_type: authorId ? 'wpcom' : 'email_only',
```

Comment:
It might also be interesting to pass through the userId here

Reply:
I'm a bit wary of logging user info, though.

Also, let's say we log the ID of registered users. What about `email_only`? Should we log the email?
I mean, this will be valuable info if we wanted to investigate users that get blocked often, but there's also a privacy concern to think of.